cdc/sink/producer/kafka/config.go

```diff
@@ -0,0 +1,297 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/pkg/config"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/security"
+	"github.com/pingcap/ticdc/pkg/util"
+	"go.uber.org/zap"
+)
+
+func init() {
+	sarama.MaxRequestSize = 1024 * 1024 * 1024 // 1GB
+}
+
+// Config stores user specified Kafka producer configuration
+type Config struct {
+	BrokerEndpoints []string
+	PartitionNum    int32
+
+	// User should make sure that `replication-factor` not greater than the number of kafka brokers.
+	ReplicationFactor int16
+
+	Version         string
+	MaxMessageBytes int
+	Compression     string
+	ClientID        string
+	Credential      *security.Credential
+	SaslScram       *security.SaslScram
+	// control whether to create topic
+	AutoCreate bool
+}
+
+// NewConfig returns a default Kafka configuration
+func NewConfig() *Config {
+	return &Config{
+		Version: "2.4.0",
+		// MaxMessageBytes will be used to initialize producer, we set the default value (1M) identical to kafka broker.
+		MaxMessageBytes:   1 * 1024 * 1024,
+		ReplicationFactor: 1,
+		Compression:       "none",
+		Credential:        &security.Credential{},
+		SaslScram:         &security.SaslScram{},
+		AutoCreate:        true,
+	}
+}
+
+// CompleteByOpts the kafka configuration.
+func (c *Config) CompleteByOpts(sinkURI *url.URL, replicaConfig *config.ReplicaConfig, opts map[string]string) error {
+	c.BrokerEndpoints = strings.Split(sinkURI.Host, ",")
+	params := sinkURI.Query()
+	s := params.Get("partition-num")
+	if s != "" {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			return err
```

Comment:
Better to use an RFC error, ditto for other parameter's Atoi error.
Besides has this error branch be tested?

Reply:
Wraped in https://github.com/pingcap/ticdc/blob/77029891ff124a5a0e07d5b4dbd4c4f40c6c6268/cdc/sink/mq.go#L384

Reply:
> Besides has this error branch be tested?

Let me test it.